dev/NavigationView/NavigationViewItem.cpp

```diff
@@ -296,6 +438,83 @@ NavigationViewItemPresenter * NavigationViewItem::GetPresenter()
     return presenter;
 }
 
+void NavigationViewItem::IsRepeaterVisible(bool visible)
+{
+    ReparentRepeater();
+
+    auto visibility = visible ? winrt::Visibility::Visible : winrt::Visibility::Collapsed;
+    m_repeater.get().Visibility(visibility);
+
+    if (ShouldRepeaterShowInFlyout() && visible)
+    {
+        winrt::FlyoutBase::ShowAttachedFlyout(m_rootGrid.get());
+    }
+}
+
+void NavigationViewItem::ReparentRepeater()
+{
+    if (auto const repeater = m_repeater.get())
+    {
+        if (ShouldRepeaterShowInFlyout() && !m_isRepeaterParentedToFlyout)
+        {
+            m_rootGrid.get().Children().RemoveAtEnd();
+            m_flyoutRootGrid.get().Children().Append(repeater);
+            m_isRepeaterParentedToFlyout = true;
+
+            PropagateDepthToChildren(0);
+        }
+        else if (!ShouldRepeaterShowInFlyout() && m_isRepeaterParentedToFlyout)
+        {
+            m_flyoutRootGrid.get().Children().RemoveAtEnd();
+            m_rootGrid.get().Children().Append(repeater);
+            m_isRepeaterParentedToFlyout = false;
+
+            PropagateDepthToChildren(1);
+        }
+    }
+}
+
+// We only want to show flyouts if the item is at the top level of the
+// item tree and navigationview is in compact mode.
+bool NavigationViewItem::ShouldRepeaterShowInFlyout()
+{
+    UpdateIsClosedCompact();
+    return (m_isClosedCompact && m_isTopLevelItem) || IsOnTopPrimary();
+}
+
+void NavigationViewItem::UpdateItemIndentation()
+{
+    // Update item indentation based on its depth
+    if (auto const presenter = m_navigationViewItemPresenter.get())
+    {
+        auto newLeftMargin = Depth() * c_itemIndentation;
+        winrt::get_self<NavigationViewItemPresenter>(presenter)->UpdateContentLeftIndentation(static_cast<double>(newLeftMargin));
+    }
+}
+
+void NavigationViewItem::PropagateDepthToChildren(int depth)
+{
+    if (auto const repeater = m_repeater.get())
+    {
+        auto itemsCount = repeater.ItemsSourceView().Count();
+        for (int index = 0; index < itemsCount; index++)
+        {
+            if (auto const element = repeater.TryGetElement(index))
+            {
+                if (auto const nvib = element.try_as<winrt::NavigationViewItemBase>())
+                {
+                    winrt::get_self<NavigationViewItemBase>(nvib)->Depth(depth);
```

Comment:
>Depth [](start = 67, length = 5)

can we do this just during ElementPrepared instead ?

Reply:
Currently we update indentation of items based on depth. We do update depth in ElementPrepared, however since we are moving the repeater between the flyout and item, the depth can become out of date if repeater was loaded in one display mode and opened in another. 
However, this may actually be a good reason to update implementation to move the data source between two repeaters instead of moving one repeater.